File: SDKs/amazonaws.com/ec2/2016-11-15/go/pkg/models/operations/postcreateegressonlyinternetgateway.go
package operations

type PostCreateEgressOnlyInternetGatewayActionEnum string

const (
	PostCreateEgressOnlyInternetGatewayActionEnumCreateEgressOnlyInternetGateway PostCreateEgressOnlyInternetGatewayActionEnum = "CreateEgressOnlyInternetGateway"
)

type PostCreateEgressOnlyInternetGatewayVersionEnum string

const (
	PostCreateEgressOnlyInternetGatewayVersionEnumTwoThousandAndSixteen1115 PostCreateEgressOnlyInternetGatewayVersionEnum = "2016-11-15"
)

type PostCreateEgressOnlyInternetGatewayQueryParams struct {
	Action  PostCreateEgressOnlyInternetGatewayActionEnum  `queryParam:"style=form,explode=true,name=Action"`
	Version PostCreateEgressOnlyInternetGatewayVersionEnum `queryParam:"style=form,explode=true,name=Version"`
}

type PostCreateEgressOnlyInternetGatewayHeaders struct {
	XAmzAlgorithm     *string `header:"style=simple,explode=false,name=X-Amz-Algorithm"`
	XAmzContentSha256 *string `header:"style=simple,explode=false,name=X-Amz-Content-Sha256"`
	XAmzCredential    *string `header:"style=simple,explode=false,name=X-Amz-Credential"`
	XAmzDate          *string `header:"style=simple,explode=false,name=X-Amz-Date"`
	XAmzSecurityToken *string `header:"style=simple,explode=false,name=X-Amz-Security-Token"`
	XAmzSignature     *string `header:"style=simple,explode=false,name=X-Amz-Signature"`
	XAmzSignedHeaders *string `header:"style=simple,explode=false,name=X-Amz-SignedHeaders"`
}

type PostCreateEgressOnlyInternetGatewayRequest struct {
	QueryParams PostCreateEgressOnlyInternetGatewayQueryParams
	Headers     PostCreateEgressOnlyInternetGatewayHeaders
	Request     []byte `request:"mediaType=text/xml"`
}

type PostCreateEgressOnlyInternetGatewayResponse struct {
	Body        []byte
	ContentType string
	StatusCode  int64
}

File: SDKs/amazonaws.com/ec2/2016-11-15/typescript/src/sdk/models/operations/postcreateegressonlyinternetgateway.ts
import { SpeakeasyMetadata, SpeakeasyBase } from "../../../internal/utils";


export enum PostCreateEgressOnlyInternetGatewayActionEnum {
    CreateEgressOnlyInternetGateway = "CreateEgressOnlyInternetGateway"
}

export enum PostCreateEgressOnlyInternetGatewayVersionEnum {
    TwoThousandAndSixteen1115 = "2016-11-15"
}


export class PostCreateEgressOnlyInternetGatewayQueryParams extends SpeakeasyBase {
  @SpeakeasyMetadata({ data: "queryParam, style=form;explode=true;name=Action" })
  action: PostCreateEgressOnlyInternetGatewayActionEnum;

  @SpeakeasyMetadata({ data: "queryParam, style=form;explode=true;name=Version" })
  version: PostCreateEgressOnlyInternetGatewayVersionEnum;
}


export class PostCreateEgressOnlyInternetGatewayHeaders extends SpeakeasyBase {
  @SpeakeasyMetadata({ data: "header, style=simple;explode=false;name=X-Amz-Algorithm" })
  xAmzAlgorithm?: string;

  @SpeakeasyMetadata({ data: "header, style=simple;explode=false;name=X-Amz-Content-Sha256" })
  xAmzContentSha256?: string;

  @SpeakeasyMetadata({ data: "header, style=simple;explode=false;name=X-Amz-Credential" })
  xAmzCredential?: string;

  @SpeakeasyMetadata({ data: "header, style=simple;explode=false;name=X-Amz-Date" })
  xAmzDate?: string;

  @SpeakeasyMetadata({ data: "header, style=simple;explode=false;name=X-Amz-Security-Token" })
  xAmzSecurityToken?: string;

  @SpeakeasyMetadata({ data: "header, style=simple;explode=false;name=X-Amz-Signature" })
  xAmzSignature?: string;

  @SpeakeasyMetadata({ data: "header, style=simple;explode=false;name=X-Amz-SignedHeaders" })
  xAmzSignedHeaders?: string;
}


export class PostCreateEgressOnlyInternetGatewayRequest extends SpeakeasyBase {
  @SpeakeasyMetadata()
  queryParams: PostCreateEgressOnlyInternetGatewayQueryParams;

  @SpeakeasyMetadata()
  headers: PostCreateEgressOnlyInternetGatewayHeaders;

  @SpeakeasyMetadata({ data: "request, media_type=text/xml" })
  request?: Uint8Array;
}


export class PostCreateEgressOnlyInternetGatewayResponse extends SpeakeasyBase {
  @SpeakeasyMetadata()
  body?: Uint8Array;

  @SpeakeasyMetadata()
  contentType: string;

  @SpeakeasyMetadata()
  statusCode: number;
}
